### PR TITLE
Fix types import/exports

### PR DESCRIPTION
**Pull request by skogsmaskin**
### Description

Fix type import/exports to work with vite.js

### CODE REVIEW

File: packages/@sanity/base/src/components/previews/types.ts [hunk 1]
```diff
@@ -11,8 +11,8 @@ export type FieldName = string
 
 export type Path = FieldName[]
 export type Selection = [Id, FieldName[]]
-export {PrepareViewOptions} from '@sanity/types'
-export {SortOrdering} from '@sanity/types'
+export type {PrepareViewOptions} from '@sanity/types'
+export type {SortOrdering} from '@sanity/types'
```
Comment (bjoerge) [inline packages/@sanity/base/src/components/previews/types.ts:15]:
```suggestion
export type {PrepareViewOptions, SortOrdering} from '@sanity/types'
```

File: packages/@sanity/base/src/preview/types.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import {Reference} from '@sanity/types'
 
-export {PrepareViewOptions} from '@sanity/types'
-export {SortOrdering} from '@sanity/types'
+export type {PrepareViewOptions} from '@sanity/types'
+export type {SortOrdering} from '@sanity/types'
```
Comment (bjoerge) [inline packages/@sanity/base/src/preview/types.ts:4]:
```suggestion
export type {SortOrdering, PrepareViewOptions} from '@sanity/types'
```

File: packages/@sanity/base/src/_exports/_internal.ts [hunk 3]
```diff
@@ -11,7 +11,8 @@ export {observePathsDocumentPair as unstable_observePathsDocumentPair} from '../
 // eslint-disable-next-line camelcase
 export {observeDocumentPairAvailability as unstable_observeDocumentPairAvailability} from '../preview/availability'
 
-export {DocumentAvailability, AvailabilityReason} from '../preview/types'
+export type {DocumentAvailability} from '../preview/types'
+export {AvailabilityReason} from '../preview/types'
```
Comment (bjoerge) [inline packages/@sanity/base/src/_exports/_internal.ts:15]:
```suggestion
export {type AvailabilityReason, DocumentAvailability} from '../preview/types'
```

**Review by bjoerge — APPROVED** [review]
Thanks! Added some suggestions, feel free to ignore as they are not really related to the problem you're fixing here 😄

**Comment by skogsmaskin** [comment]
> Thanks! Added some suggestions, feel free to ignore as they are not really related to the problem you're fixing here smile

I have added your suggestions and squashed them into my commit.

File: packages/@sanity/base/src/_exports/_internal.ts (hunk 3 above)
Comment (skogsmaskin) [reply packages/@sanity/base/src/_exports/_internal.ts]: This one doesn't seem like a valid way to do the export.

Comment (bjoerge) [reply packages/@sanity/base/src/_exports/_internal.ts]: it is valid typescript, but maybe other parsers haven't caught up yet

Comment (skogsmaskin) [reply packages/@sanity/base/src/_exports/_internal.ts]: Must be `export type {AvailabilityReason, DocumentAvailability} from '../preview/types'`

**Pull request merged**